File: CPP/LeftAndRightSumDifferences.cpp
#include <cstdlib>
#include <iostream>
#include <numeric>
#include <vector>

class Solution {
public:
    std::vector<int> leftRightDifference(std::vector<int>& nums) {
        int total = std::accumulate(nums.begin(), nums.end(), 0);
        std::vector<int> ans(nums.size());
        int left = 0;
        for (size_t i = 0; i < nums.size(); ++i) {
            int right = total - left - nums[i];
            ans[i] = std::abs(left - right);
            left += nums[i];
        }
        return ans;
    }
};

void print(const std::vector<int>& v) {
    std::cout << "[";
    for (size_t i = 0; i < v.size(); ++i) {
        if (i) std::cout << ", ";
        std::cout << v[i];
    }
    std::cout << "]\n";
}

int main() {
    Solution s;
    std::vector<int> a = {10, 4, 8, 3};
    std::vector<int> b = {1};
    print(s.leftRightDifference(a));  // [15, 1, 11, 22]
    print(s.leftRightDifference(b));  // [0]
    return 0;
}

File: Python/LeftAndRightSumDifferences.py
from typing import List


class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        # Single pass: left accumulates the prefix; right = total - left - nums[i].
        total = sum(nums)
        left = 0
        ans = []
        for x in nums:
            right = total - left - x
            ans.append(abs(left - right))
            left += x
        return ans


if __name__ == '__main__':
    s = Solution()
    print(s.leftRightDifference([10, 4, 8, 3]))  # [15, 1, 11, 22]
    print(s.leftRightDifference([1]))            # [0]
